src/main/scala/com/abdulradi/opentracing/xray/v1/format/http.scala

```diff
@@ -0,0 +1,30 @@
+package com.abdulradi.opentracing.xray.v1.format
+
+import com.abdulradi.opentracing.xray.v1.model._
+import eu.timepit.refined.string.Url
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.semiauto._
+
+
+object http {
+  implicit val segmentHttpEncoder: Encoder[SegmentHttp] = deriveEncoder[SegmentHttp]
+  implicit val segmentHttpDecoder: Decoder[SegmentHttp] = deriveDecoder[SegmentHttp]
+
+  implicit val subSegmentHttpEncoder: Encoder[SubsegmentHttp] = deriveEncoder[SubsegmentHttp]
+  implicit val subSegmentHttpDecoder: Decoder[SubsegmentHttp] = deriveDecoder[SubsegmentHttp]
+
+  implicit val segmentRequestEncoder: Encoder[SegmentRequest] = deriveEncoder[SegmentRequest]
+  implicit val segmentRequestDecoder: Decoder[SegmentRequest] = deriveDecoder[SegmentRequest]
+
+  implicit val subSegmentRequestEncoder: Encoder[SubsegmentRequest] = deriveEncoder[SubsegmentRequest]
```

Comment:
This encoder can't be automatically derived because the case class structure is not similar to JSON model. Fields like `method`, `url`, `user_agent`, and `client_ip` are grouped under a member called `commonFields`.

Reply:
As a rule of thumb, any class that has a member that ends with `Fields` can't be automatically derived. There is exactly 9 occurrences in the code. Search for "Fields:" you should find 2 occurrences in http.scala and 7 in segment.scala.